### PR TITLE
Add average market change line to equity plot

### PR DESCRIPTION
**Pull request by WillemGerritzen**
# What has been changed?

Added a trace to the equity plot tracking the average market change for all coins 

# Potential Issues / Future considerations

The computation of the average starting capital assumes an equal division between all coins. This will need to be refactored once we allow users to specify how to allocate the funds.

# Assumptions made / Things to add to the wiki
 
Might be good to add (if not already the case) in the docs that plotly allows the traces to be toggled by clicking on their respective legends

### CODE REVIEW

File: modules/output/equity_plot.py [hunk 1]
```diff
@@ -1,29 +1,39 @@
...
-            inplace=True)
+    df_capital = pd.DataFrame(list(stats.capital_per_timestamp.items()), columns=['timestamp', 'capital'])
+
+    df_capital, dates = convert_df_for_plotting(df_capital)
 
-    # get dates and y range
-    dates = [datetime.fromtimestamp(time / 1000) for time in df[0]]
-    min_value = max(int(df[1].min() - 10), 1)
-    max_value = int(df[1].max() + 10)
+    df_coins = compute_average_market_change(stats.df, stats.capital_per_timestamp[0])
```
Comment (sheridavandenbent) [inline modules/output/equity_plot.py:17]: Be mindful of your variables - while it's clear to you that df_coins means the average market change over all the coins normalized by the starting capital, that might not be clear to other people reading your code.

How about 'avg_market_change_coins'? (Be sure to change it elsewhere in the code too

```suggestion
    avg_market_change_coins = compute_average_market_change(stats.df, stats.capital_per_timestamp[0])
```

File: modules/output/equity_plot.py [hunk 2]
```diff
@@ -1,29 +1,39 @@
...
+    data = [
+        {
+            "name": "Capital",
+            "mode": "lines",
+            "x": dates,
+            "y": df_capital['capital'],
+            "fill": 'tozeroy'
+        },
+        {
+            "name": "Average market change",
```
Comment (sheridavandenbent) [inline modules/output/equity_plot.py:29]:
```suggestion
            "name": "Average Market Change",
```

It looks better when it's all capitalised, imho :)
However, this is very much nitpicking on my side, so if you don't agree feel free to disregard this.

File: modules/output/plots.py [hunk 3]
```diff
@@ -84,3 +88,40 @@ def add_indicators(fig, dates, df, mainplot_indicators, subplot_indicators):
...
+    Simulates a buy at the first candle and no selling to be plotted on the equity plot
+    """
+
+    coins = [coin for coin in df_stats]
+    df_coins = pd.DataFrame(columns=coins)
+
+    # Compute average starting capital per coin assuming it's equally divided among all coins
+    avg_starting_capital = starting_capital / len(coins)
+
+    for coin, ohlcv in df_stats.items():
```
Comment (sheridavandenbent) [inline modules/output/plots.py:120]: This all looks really neat. Well done on restructuring this!

**Review by sheridavandenbent — CHANGES_REQUESTED** [review]
Overall really good - I have one small feedback point on the naming of variables, but that's it. Great job on this one 👍

File: modules/output/plots.py (hunk 3 above)
Comment (WillemGerritzen) [reply modules/output/plots.py:120]: Thanks!

**Review by sheridavandenbent — APPROVED** [review]
Looks great!

**Pull request merged**